charset.go

```diff
@@ -49,7 +49,16 @@ func charsetReader(charset string, input io.Reader) (io.Reader, error) {
 // decodeHeader decodes an internationalized header field. If it fails, it
 // returns the input string and the error.
 func decodeHeader(s string) (string, error) {
-	wordDecoder := mime.WordDecoder{CharsetReader: charsetReader}
+	charsetReaderWrapper := func(charset string, input io.Reader) (io.Reader, error) {
```

Comment:
Actually, it would probably make more sense to wrap the error with `UnknownCharsetError` in our `charsetReader` function right above? That way we wouldn't need to do it at each call site.

Reply:
Yeah, that's better, done